pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -164,12 +164,12 @@ func (nm *NodeManager) DiscoverNode(node *v1.Node) error {
 				defer cancel()
 				vm, err := res.datacenter.GetVMByUUID(ctx, nodeUUID)
 				if err != nil {
-					glog.V(4).Infof("Error %q while looking for vm=%+v in vc=%s and datacenter=%s",
+					glog.V(4).Infof("Error %q while looking for node %s vm=%+v in vc=%s and datacenter=%s",
 						err, node.Name, vm, res.vc, res.datacenter.Name())
 					if err != vclib.ErrNoVMFound {
 						setGlobalErr(err)
 					} else {
-						glog.V(4).Infof("Did not find node %s in vc=%s and datacenter=%s",
+						glog.V(4).Infof("Did not find node %s in vc=%s and datacenter=%s, err: %v",
```

Comment:
Not logging the error here seemed intentional. The only error that this could be is `vclib.ErrNoVMFound`, which the log message does a good job of explaining. Is there particular context in the error that's missing from the log?

Reply:
Sorry, my mistakes. we should delete 'err' in `glog.V(4).Infof("Did not find node %s in vc=%s and datacenter=%s", node.Name, res.vc, res.datacenter.Name(), err)`, rather than add  pattern `err: %v`.